micropython/bluetooth/aioble/aioble/server.py

```diff
@@ -97,6 +102,33 @@ def write(self, data, send_update=False):
         else:
             ble.gatts_write(self._value_handle, data, send_update)
 
+    async def _init_shared_write_task(self):
+        # Should be run on first call of an async task waiting on .written()
+        # Will create the needed events and start the background queue monitor task
+        if BaseCharacteristic._shared_write_event is None:
```

Comment:
Might be possible to change this to `hasattr(...)` and then remove the `..=None` settings in the global class scope above.

Reply:
I often struggle with questions like this when it's a bit of a ram vs runtime cost, however in this case yeah - the current ram cost of having that attr "preallocated" is paid by people not using the feature, whereas a hasattr call vs "attr lookup and compare to none" would be nigh on identical runtime cost. I'll make the change thanks!